### PR TITLE
Add drinking water quest

### LINKED ISSUE #549: New quest: Is water drinkable? (drinking_water)
[drinking_water](https://wiki.openstreetmap.org/wiki/Key:drinking_water) seems to be promising for a quest.

* [ ] Easy to answer, i.e. everyone can easily solve it (from the outside)
* [x] Easy user interface. (quite easy, depends on how much details to be implement)
* [x] Has a useful purpose. (important information)
* [x] Established tag. (seems to be; ~50 000 times used)
* [x] Low unanswerable quests. (i.e. low false positives, as every quest can be answered; at least if there is a sign)
* [x] It can be determined, if the quest is needed to be asked. (i.e. there is drinking_water=no)

It should be asked for [the things mentioned here](https://wiki.openstreetmap.org/wiki/Key:drinking_water#Applies_to_.28among_other_things.29), maybe only exclude toilets (they are too common?) and some other tags, where it is not alwayss clear that there is water at all (i.e. rest_area/camp_site).

### PR DESCRIPTION
**Pull request by FloEdelmann**
Closes #549.

### CODE REVIEW

File: app/src/main/java/de/westnordost/streetcomplete/quests/drinking_water/AddDrinkingWater.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    override val elementFilter = """
+        nodes with (
+            man_made = water_tap
+            or man_made = water_well
+            or amenity = toilets
+            or natural = spring
+        )
+        and access !~ private|no and indoor != yes
+        and !drinking_water and amenity != drinking_water
```
Comment (FloEdelmann) [inline app/src/main/java/de/westnordost/streetcomplete/quests/drinking_water/AddDrinkingWater.kt:17]: Maybe even exclude any nodes with `drinking_water:legal`?

```suggestion
        and !drinking_water and !drinking_water:legal and amenity != drinking_water
```

**Comment by westnordost** [comment]
Nice initiative! Also, nice icon! 

But phew, of all the other quest suggestions you pick that one to implement! If I remember right from the linked ticket, the discussion around this quest ended in a limbo, with all that `drinking_water:legal` and stuff.
Could you summarize your position on this and how you would solve (/solved) the issues mentioned in that ticket?

**Comment by FloEdelmann** [comment]
> Nice initiative! Also, nice icon!

Thanks :blush: 

> But phew, of all the other quest suggestions you pick that one to implement!

It seemed like the easiest for me to implement :sweat_smile: 

> Could you summarize your position on this and how you would solve (/solved) the issues mentioned in that ticket?

I tried to make the options clear through wording:

* If it's signed, it's easy to answer.
* If it's not signed, and easy to decide (like https://goo.gl/K2hp3W vs. https://goo.gl/PKuP4U), it's easy to answer, too.
* If it's not signed, and difficult to decide, users might tap on "Other answers" and choose "seems drinkable", or skip the quest (with or without a note).
* If it's signed, but clearly wrong (e.g. in restaurant toilets etc., or by applying local knowledge), none of the answers are applicable, so users will tap on "Other answers" → "Sign is wrong", and leave a note. Then it will be decided case by case.

File: app/src/main/java/de/westnordost/streetcomplete/quests/drinking_water/AddDrinkingWater.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    """
+
+    override val commitMessage = "Add whether water is drinkable"
+    override val wikiLink = "Key:drinking_water"
+    override val icon = R.drawable.ic_quest_drinking_water
+    override val isDeleteElementEnabled = true
```
Comment (FloEdelmann) [inline app/src/main/java/de/westnordost/streetcomplete/quests/drinking_water/AddDrinkingWater.kt:23]: Not sure if this is necessary in this quest.

**Comment by westnordost** [comment]
Okay, I am sorry, I thought I wrote something on the #549 ticket before, but I did not.

Let's do the following:
Throw the uncommon answers and/or answers that require explanation all into one bucket, into the "Can't say" bucket:
    - It's better to leave a note (with picture) than to delete the node if the node can't be found because as you say, it might be a faucet in a toilet or something like that
    - sign is wrong: Better have a note+photo here, cause this may be quite subjective
    - seems drinkable: This answer is too close to just simply "yes, but no sign" anyway. If the surveyor wants to express that he is not 100% sure, better let him leave a note

Regarding the answer options remaining, I am not completely happy about these. At least, the wording is too long. Also, it should avoid "yes" and "no" because there are not 2 but 4 answer options. Maybe:

"Potable, stated by a sign"
"Non-potable, stated by a sign"
"Potable, obvious from the context"
"Non-potable, obvious from the context"

Still a lot of text. Maybe (additionally) use pictures? 
image

Are really all answer options realistic and necessary? I.e. does none of the options require explanation (-> create note) and none of the options is fuzzy enough that users can not always be sure about it (-> create note)?

File: app/src/main/java/de/westnordost/streetcomplete/quests/drinking_water/AddDrinkingWater.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.osmquest.OsmFilterQuestType
+
+class AddDrinkingWater : OsmFilterQuestType<DrinkingWater>() {
+
+    override val elementFilter = """
+        nodes with (
+            man_made = water_tap
+            or man_made = water_well
+            or amenity = toilets
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/quests/drinking_water/AddDrinkingWater.kt:13]: should remove that. Too common and it will be always the same answer that depends on the region. ("Yes, no sign" or "no, no sign")

File: app/src/main/java/de/westnordost/streetcomplete/quests/drinking_water/AddDrinkingWater.kt (hunk 1 above)
Comment (westnordost) [reply app/src/main/java/de/westnordost/streetcomplete/quests/drinking_water/AddDrinkingWater.kt:17]: yes. If only `drinking_water:legal` is set but not `drinking_water`, it will have a good reason

File: app/src/main/java/de/westnordost/streetcomplete/quests/drinking_water/AddDrinkingWater.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        and access !~ private|no and indoor != yes
+        and !drinking_water and amenity != drinking_water
+    """
+
+    override val commitMessage = "Add whether water is drinkable"
+    override val wikiLink = "Key:drinking_water"
+    override val icon = R.drawable.ic_quest_drinking_water
+    override val isDeleteElementEnabled = true
+
+    override fun getTitle(tags: Map<String, String>) = R.string.quest_drinking_water_title
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/quests/drinking_water/AddDrinkingWater.kt:25]: I think it makes sense to have a different question for the different things to ask for. I.e. the user should know if (s)he is looking for a faucet, or a natural spring, or a water well.

However, we could try something different here.

We have the feature dictionary. If all of man_made = water_tap, man_made = water_well etc. are in the id presets, the question could also appear like this:

`Is the water drinkable here? (Natural spring)`

This helps to keep down the number of strings that must be translated.

Comment (westnordost) [reply app/src/main/java/de/westnordost/streetcomplete/quests/drinking_water/AddDrinkingWater.kt:25]: Whether a preset for a thing exists, can be checked by searching through the `presets.json` (not the brands one)

Comment (FloEdelmann) [reply app/src/main/java/de/westnordost/streetcomplete/quests/drinking_water/AddDrinkingWater.kt:25]: They are indeed all included in `presets.json`:

* [`man_made=water_tap`](https://github.com/openstreetmap/id-tagging-schema/blob/9fdecc983443ce75185c2923caae9f7703f2cae4/dist/presets.json#L15735)
* [`man_made=water_well`](https://github.com/openstreetmap/id-tagging-schema/blob/9fdecc983443ce75185c2923caae9f7703f2cae4/dist/presets.json#L15768)
* [`natural=spring`](https://github.com/openstreetmap/id-tagging-schema/blob/9fdecc983443ce75185c2923caae9f7703f2cae4/dist/presets.json#L16297)

How can I access those from the code?

Comment (westnordost) [reply app/src/main/java/de/westnordost/streetcomplete/quests/drinking_water/AddDrinkingWater.kt:25]: See for example `AddPlaceName.kt`, `getTitle` and `getTitleArgs`

Comment (westnordost) [reply app/src/main/java/de/westnordost/streetcomplete/quests/drinking_water/AddDrinkingWater.kt:25]: I think `arrayOf(featureName.value.toString())` should be fine too. In the worst case (-> somehow that preset was removed), the question would appear as `Is the water drinkable here? (null)`

**Comment by FloEdelmann** [comment]
> Throw the uncommon answers and/or answers that require explanation all into one bucket, into the "Can't say" bucket

Okay, done.

> "Potable, stated by a sign"
> "Non-potable, stated by a sign"
> "Potable, obvious from the context"
> "Non-potable, obvious from the context"

Those are fine for me, too.

Regarding *potable* vs. *drinkable*: *Potable* seems to be a bit more technically correct ("safe to drink, free from poisons etc.") and also more widespread. However, *drinkable* is also not uncommon, seems to be used interchangeably in colloquial language, and maybe might make it easier for non-native speakers to understand (at least I've never heard *potable* before). On the other hand, the roman languages all seem to use a word similar to *potable*.

> Maybe (additionally) use pictures?

Those limit the possible text length even more (especially in translations). Also, is it clear from the accompanying description that the sign does not have to look exactly like that? There is a wide variety of "drinking water" / "no drinking water" signs, see https://www.ecosia.org/images?q=trinkwasser%20schild and https://www.ecosia.org/images?q=drinking%20water%20sign

> Are really all answer options realistic and necessary? I.e. does none of the options require explanation (-> create note) and none of the options is fuzzy enough that users can not always be sure about it (-> create note)?

There exist both "drinking water" and "no drinking water" signs, so the first two options are realistic and necessary. When there is a sign, the user can assume that it's correct, so no note is necessary.

Only in the case that the user has local knowledge that the sign is wrong, they might be tempted to choose one of the "obvious from the context" answers instead of leaving a note. I think that case is better handled by my original wording.

When there is no sign, in many cases it's obvious (e.g. [here](https://goo.gl/K2hp3W), [here](https://goo.gl/PKuP4U) or [here](http://cdn.onlyinyourstate.com/wp-content/uploads/2016/02/Ice_Cold_Spring_Water-01.jpg)), but of course there are also cases where it's not obvious (e.g. [here](https://s3.wasabisys.com/saudiawebdesigncompany/2017/09/garden-faucet-design-ideas-faucet-extension-lee-valley-tools-regarding-dimensions-957-x-1300.jpg) or [here](https://wv-oberdiessbach.ch/schwand%20aussen.JPG)). I think then, none of the options are fitting enough to choose them instead of leaving a note.

**Comment by FloEdelmann** [comment]
What about this wording?

**Is the water potable here? (%s)**

* Potable, stated by a sign
* Non-potable, stated by a sign
* Clearly potable (no sign)
* Clearly non-potable (no sign)

**Comment by westnordost** [comment]
I think we need some native speakers on this. @peternewman @smichel17 ?

**Comment by smichel17** [comment]
- "Potable" is the correct term. In my *opinion* everyone *should* know it, but in reality I am not not sure everyone *will*. So, I would also include "safe to drink".
- I have no opinion on whether it is better to omit "potable" entirely or use a mix of both, like in the example below.
- I have no opinion on which parts use which; they are freely interchangeable.
- You may wish to include a note to translators that they can translate either way.
    - One option: always use potable in the source set and then provide an en_US, translation which uses "safe to drink"
- You could provide a graphic but I would be careful not to imply anything unintentional — when there are images, people tend to skip over the words.
    - In particular, I'm worried the faucet icon might be misinterpreted as "This question asks about the type of water source". Not such a big deal for the quest icon (unless you will have a "what type of water source is this" quest later, in which case, consider using this icon for that), but if you put images in the answer, perhaps a water fountain or emoji-drinking-from-a-cup would be a better option.
- (signed) vs (unsigned) is clearest, in my opinion.

> ### Is this water source safe to drink?
> - [ ] Potable (signed)
> - [ ] Potable (unsigned)
> - [ ] Not potable (signed)
> - [ ] Not potable (unsigned)

**Comment by peternewman** [comment]
Potable may be used on some signs and I'm aware of it from camping etc, but as @smichel17 says, I wonder if it's too technical. I think drinkable/safe-to-drink may be more widely understood.

I like @smichel17 's suggestion, it seems to cover all the boxes, as long as you don't think potable means unsafe to drink (which I guess the picture(s) would help with)...

What's wrong with the horse trough? In the UK that would probably be fed by mains pressure water, so is probably safer than drinking out of a stream, yet I've done the latter but not the former... :laughing: Maybe not while the horse is there too, although with that one having a hose rather than a ball cock, you could pull it out and use it pretty cleanly.

It's also interesting looking at your signs that the German is clearly drinking water, where as the French go with potable.

**Comment by smichel17** [comment]
> as long as you don't think potable means unsafe to drink

I was flip-flopping about which way it should be, but you're right, the way I put it is ambiguous; swapping would solve that problem (while retaining the technical word, so that anyone who knows it, knows exactly what they're mapping):

> ### Is this water source potable?
> - [ ] Safe to drink (signed)
> - [ ] Safe to drink (unsigned)
> - [ ] Not safe to drink (signed)
> - [ ] Not safe to drink (unsigned)

It is a bit lengthier, but I think short enough in English, and other translations can use that language's equivalent of potable if they need the space.

> I think drinkable/safe-to-drink may be more widely understood.

I think drinkable is too vague. Most people will read it as "safe to drink", but I can imagine some people reading it as "possible to drink", and answer that river water which tastes fine but is not safe (due to bacteria) is "drinkable".

**Comment by westnordost** [comment]
What's a ball cock?

**Comment by westnordost** [comment]
I think it is best to both mention "safe to drink" and "potable" in the form, so that it is clear that it is pretty synonymous. "Potable" is also shorter.

**Comment by peternewman** [comment]
> I was flip-flopping about which way it should be, but you're right, the way I put it is ambiguous; swapping would solve that problem (while retaining the technical word, so that anyone who knows it, knows exactly what they're mapping):

That's perfect!

> > I think drinkable/safe-to-drink may be more widely understood.
> 
> I think drinkable is too vague. I think most people will read it as "safe to drink", but some may read it as "possible to drink", and answer that river water which tastes fine but is not safe (due to bacteria) is "drinkable", or something like that.
> "Safe to drink" is still a little ambiguous, because or varying tolerance to risk, but much less so.

Yeah fair point. I was more meaning in comparison to potable.

> What's a ball cock?

Nothing rude, just a float valve, used in traditional UK toilets too. Horse drinks water, water level drops, float valve opens, water trough refills. I can't see anything like that in that image, hence my confusion. This isn't a common UK design but it shows the operation:
https://www.jamessmithfencing.co.uk/product/concrete-water-trough/concrete-water-trough-75-gallon/

> > It's also interesting looking at your signs that the German is clearly drinking water, where as the French go with potable.
> 
> Where, on the wiki?

No, @FloEdelmann 's image searches:
https://www.ecosia.org/images?q=trinkwasser%20schild and https://www.ecosia.org/images?q=drinking%20water%20sign

> I think it is best to both mention "safe to drink" and "potable" in the form, so that it is clear that it is pretty synonymous. "Potable" is also shorter.

I think @smichel17 does that quite elegantly, or do you think it needs to say "potable (safe to drink)"?

**Comment by FloEdelmann** [comment]
Thanks for your input!

I've changed the wording to @smichel17's second suggestion:

**Is this water source potable? (*Natural Spring*)**

- Safe to drink (signed)
- Safe to drink (unsigned)
- Not safe to drink (signed)
- Not safe to drink (unsigned)

> You may wish to include a note to translators that they can translate either way.

The strings are now called `quest_drinking_water_potable_signed` etc.
Do you think that implicit hint is sufficient enough for translators?

**Comment by smichel17** [comment]
I don't know how it appears in POEditor but assuming the string name is visible, I think that it is probably good enough. But it's also hard for me to know, since I've thought about it, vs seeing it for the first time… but also if a direct translation of "potable" is well known in that language, I would somewhat expect them to use that word anyway, as opposed to a word-by-word translation of "safe to drink". So yeah, probably good enough. If you really want, you could send one string to a translator and ask them to try translating it and then link to this conversation and ask if the flexibility in phrasing was understood :shrug:

**Comment by FloEdelmann** [comment]
> I don't know how it appears in POEditor

It does:

**Comment by matkoniecz** [comment]
> It seemed like the easiest for me to implement sweat_smile

Obviously, I may be wrong but #2507 looks nice (reuse of existing code or maybe even simple yes/no quest), well researched, propably has no tagging traps.

**Comment by FloEdelmann** [comment]
So what is left to do here?

* I believe the wording is as simple, concise and short as possible now.
* Translators have an additional hint in the string names (which are shown in POEditor).
* All fuzzy options have been removed in favor of the "Can't say" option.
* The "signed" options are easy to determine and have a clear tagging scheme.
* The "unsigned" options are more fuzzy and depend on the user's judgement. I think that most false conclusions are prevented by the wording though.

The quest does still need an achievement category. Is it more fitting for "outdoors" or "citizen"?

**Comment by matkoniecz** [comment]
> The quest does still need an achievement category. Is it more fitting for "outdoors" or "citizen"?

outdoors to me, though quest can be in two categories (at least one is already)

**Review by westnordost — APPROVED** [review]

**Comment by westnordost** [comment]
Yeah, I'd add it to outdoors

**Review by westnordost — APPROVED** [review]
